subsys/modem/modem_cmux.c

```diff
@@ -816,6 +816,14 @@ static void modem_cmux_process_received_byte(struct modem_cmux *cmux, uint8_t by
 		/* Get last 8 bits of data length */
 		cmux->frame.data_len |= ((uint16_t)byte) << 7;
 
+		if (cmux->frame.data_len > cmux->receive_buf_size) {
+			LOG_ERR("Frame data length %u exceeds receive buffer size %u",
```

Comment:
Could you add "indicated frame data length" or something like that to make it clear its not the actual amount of data exceeding the buffer, but the expected value indicated by the frame header?

Reply:
Yes! absolutely